### DISCUSSION TITLE
HOGWARTS: Trouble getting Form Alert to render on DOM

### DISCUSSION THREAD

**Discussion by elliewalkersm** [nss-evening-cohort-14/evening-client-side#83] [Help Me!] [answered]
Ellie Walker

Have you searched the other help tickets?
Yes

Describe your issue:
I am working on adding the alert message when the name input field is left blank. I am not able to get it to work just yet.

List any error messages that appear:
none

What have you googled?
https://www.w3schools.com/js/js_popup.asp
https://getbootstrap.com/docs/5.0/components/alerts/
https://stackoverflow.com/questions/27284934/how-to-alert-user-for-blank-form-fields

What have you tried?
I tried creating a formAlert function and passing it through an if statement in my formContent function. I also tried moving the formAlert function inside the my formContent function. I also tried eliminating the formAlert function completely and just adding alert('please enter a name'); inside the if statement. 

Copy/paste link TO THE FILE on your branch or in your repo:
https://github.com/elliewalkersm/sorting-hat/blob/form-alert/main.js

**Comment by TrinityChristiana** [comment]
Try changing the type of this button to submit and let me know if that works!
https://github.com/elliewalkersm/sorting-hat/blob/form-alert/main.js#L22

Change this
```js
      <button type="button" class="btn btn-secondary" id="sortBtn">Sort!</button>
```
to this:
```js
      <button type="submit" class="btn btn-secondary" id="sortBtn">Sort!</button>
```

**Reply by elliewalkersm** [reply]
Still not working

**Comment by elliewalkersm** [comment, marked as answer]
I set an empty div on my HTML with the id=formAlert and then targeted that id in printToDom in my formAlert function.

```  const formAlert = () => {
    let domString =
      `<div class="alert alert-warning alert-dismissible fade show" role="alert">
         Please enter a name
        <button type="button" class="btn-close" data-bs-dismiss="alert" aria-label="Close"></button>
      </div>`;
      printToDom('#formAlert', domString);
  };
```

**Reply by tsekeres** [reply]
Same problem. But now how to make it disappear after successful submit??
